pkg/scheduler/plugins/resource-strategy-fit/resource_strategy_fit.go

```diff
@@ -137,6 +151,48 @@ func (rsf *resourceStrategyFitPlugin) OnSessionOpen(ssn *framework.Session) {
 	}
 }
 
+func isValidWildcardPattern(pattern string) bool {
+	if pattern == "*" {
+		return false
+	}
+
+	if !strings.HasSuffix(pattern, "*") {
+		return false
+	}
+
+	asteriskCount := strings.Count(pattern, "*")
+	if asteriskCount != 1 {
+		return false
+	}
+
+	lastIndex := strings.LastIndex(pattern, "*")
+	return lastIndex == len(pattern)-1
+}
+
+func findResourceConfigWithPrefix(resourceName string, resources map[v1.ResourceName]ResourcesType) (ResourcesType, bool) {
+	if config, exists := resources[v1.ResourceName(resourceName)]; exists {
+		return config, true
+	}
+	var bestMatch string
+	var bestConfig ResourcesType
+	var found bool
+
+	for configKey, config := range resources {
+		configKeyStr := string(configKey)
+		if strings.HasSuffix(configKeyStr, "*") {
```

Comment:
The previous one has been verified by `isValidWildcardPattern`, so there is no need to add * here. We should put all the format validation logic in `isValidWildcardPattern`